torch/_inductor/codegen/cpp.py

```diff
@@ -1547,6 +1547,8 @@ def store(self, name, index, value, mode=None):
             if not config.cpp.dynamic_threads and self.num_threads == 1:
                 line = f"{var}[{cexpr_index(index)}] += {value};"
             else:
+                if value.dtype in [torch.float16]:
```

Comment:
Shouldn't this be:
```suggestion
                if V.graph.get_dtype(name) == torch.float16:
```
?

Reply:
and leave a comment, given that if anyone updates the behaviour in the load, we should update this.

Reply:
cc @jgong5

Reply:
Now using unconditional casting.